src/lib/run-studies.ts

```diff
@@ -144,24 +97,29 @@ const main = async (): Promise<void> => {
 
     const result = await managementAppGetRunnableStudiesRequest()
     const toaGetRunsResult = await toaGetRunsRequest()
+    const existingAwsRuns: string[] = []
+
+    for (const run of result.runs) {
+        if (await checkRunExists(taggingClient, run.runId)) {
+            existingAwsRuns.push(run.runId)
+        }
+    }
 
-    const filteredResult = filterManagmentAppRuns(result, toaGetRunsResult)
+    const filteredResult = filterManagmentAppRuns(result, toaGetRunsResult, existingAwsRuns)
```

Comment:
Just a nitpick here, if `filterManagementAppRuns` has `result`, im wondering if you could just take the above block and do that inside the method? that would remove the need for the third param
```
    const existingAwsRuns: string[] = []

    for (const run of result.runs) {
        if (await checkRunExists(taggingClient, run.runId)) {
            existingAwsRuns.push(run.runId)
        }
    }
 ```

Reply:
Just to be sure I understand, are you thinking something like the following invocation where the calls to `checkRunExists` are then inside of `filterManagmentAppRuns`?:

```js
const filteredResult = filterManagmentAppRuns(result, toaGetRunsResult)
```

If so, my main reasons for keeping that logic out are:

* So the scope of `filterManagmentAppRuns`` is super simple and unaware of how we determine whether a run exists in AWS. 
    * For example, I think with your suggestion we'd still have to pass `taggingClient` to `filterManagmentAppRuns`, and so things quickly get "leaky"
* The current loop for how we check AWS for runs is probably not the most efficient and / or the specifics may change later per [this comment](https://openstax.atlassian.net/browse/SHRMP-31?focusedCommentId=13159). I didn't want to over think it for now, though, and figured no matter what, with this approach, `filterManagmentAppRuns` wouldn't be impacted anyways :man_shrugging: .

Reply:
totally - makes sense! fine keeping it as is too :)

Reply:
id say lets go ahead and merge this 👍